Lib/test/test_inspect/test_inspect.py

```diff
@@ -3545,6 +3648,169 @@ def __init__(self, b):
                            ('bar', 2, ..., "keyword_only")),
                           ...))
 
+    def test_signature_on_class_with_init(self):
+        class C:
+            def __init__(self, b):
+                pass
+
+        C(1)
```

Comment:
I understand that this is to test whether the method works without an exception, but this statement is a bit weird by itself. What are the possible use cases for a `@staticmethod` `__init__`? To be honest all the decorators on `__init__` and `__new__` are so weird that I doubt if there's anyone actually using them.

Reply:
They are weird because they are the only examples that can be created using existing builtin classes. But in future we could implement `__init__`, `__new__` and `__call__` in C for some classes. By testing with different types of callables and descriptors we increase confidence in working with arbitrary appropriate types.

Reply:
For the first sentence I actually meant the `C(1)` statement - it does not explicitly check against anything. I understand that it tests whether an exception will be raised when instantiating the object, but it still looks a bit weird.